python/cuml/utils/input_utils.py

```diff
@@ -248,3 +248,95 @@ def order_to_str(order):
         return 'column'
     elif order == 'C':
         return 'row'
+
+
+def input_to_host_array(X, order='F', deepcopy=False,
+                        check_dtype=False, convert_to_dtype=False,
+                        check_cols=False, check_rows=False,
+                        fail_on_order=False):
+    """
+    Convert input X to host array (NumPy) suitable for C++ methods that accept
+    host arrays.
+    Acceptable input formats:
+    * Numpy array - returns a pointer to the original input
+    * cuDF Dataframe - returns a deep copy always
+    * cuDF Series - returns by reference or a deep copy depending on
+        `deepcopy`
+    * cuda array interface compliant array (like Cupy) - returns a
+        reference unless deepcopy=True
+    * numba device array - returns a reference unless deepcopy=True
+
+    Returns: namedtuple('host_array', 'array pointer n_rows n_cols dtype')
```

Comment:
Should add a full docstring for Parameters... maybe it should be copied from the device_array version?